vyper/context/types/function.py

```diff
@@ -352,6 +353,21 @@ def from_AnnAssign(cls, node: vy_ast.AnnAssign) -> "ContractFunction":
             state_mutability=StateMutability.NONPAYABLE,
         )
 
+    @property
+    def canonical_type(self) -> str:
```

Comment:
Not sure if this is a meaningful method for a function to implement? It shouldn't be used in ABI encoding as a type :thinking:

Reply:
It's just an intermediate step on the way to `method_id`.  It could have a different name, or we could handle the entire logic within `method_id`.

Reply:
I think for this node type it may not make sense to have this, should probably throw `UnimplementedError` instead